### PR TITLE
fix Graph memory attribute set accounting

### LINKED ISSUE #2154: Memory overreporting for Graphs with nodes with many lables
Attribute set sizes are improperly accounted for by GRAPH.MEMORY, resulting in over reporting in graphs where nodes have many overlapping labels.

### PR DESCRIPTION
**Pull request by GomezGab**
Closes #2154

## Summary by CodeRabbit

* **New Features**
  * `GRAPH.MEMORY USAGE` now reports aggregated totals for attribute memory: total node attributes and total edge attributes.

* **Bug Fixes**
  * Attribute-memory reporting was revised to remove the older per-category breakdowns, producing more consistent total sizing.

* **Tests**
  * Updated memory-usage tests to validate the new node/edge attribute totals for empty, node-only, and multi-edge cases, including tighter tolerance expectations.

### CODE REVIEW

File: src/graph/graph_memoryUsage.h [hunk 1]
```diff
@@ -16,6 +16,8 @@ typedef struct {
 	size_t node_block_storage_sz;   // node memory usage
 	size_t edge_block_storage_sz;   // edge memory usage
 	size_t total_graph_sz_mb;       // total size in MB
+	size_t total_node_attr_sz;      // total node attribute-set memory usage
+	size_t total_edge_attr_sz;      // total edge attribute-set memory usage
```
Comment (swilly22) [inline src/graph/graph_memoryUsage.h]: isn't these two redundant ?
can't the user just add up `node_attr_by_label_sz` and `unlabeled_node_attr_sz` to get `total_node_attr_sz` ?

File: src/commands/cmd_memory.c [hunk 2]
```diff
@@ -88,9 +88,13 @@ static void _Graph_Memory
 	//    }
 	//
 	//    indices_sz_mb: <indices_sz_mb>
+	//
+	//    total_node_attributes_sz_mb: <total_node_attributes_sz_mb>
+	//
+	//    total_edge_attributes_sz_mb: <total_edge_attributes_sz_mb>
```
Comment (swilly22) [inline src/commands/cmd_memory.c]: Can't this become a source for confusion?
Users might expect `total_node_attributes_sz_mb` to equal `amortized_node_attributes_by_label_sz_mb` + `amortized_unlabeled_nodes_sz_mb`

**Review by swilly22 — COMMENTED** [review]
Let's discuss my comments to decide how we want to proceed.

File: src/commands/cmd_memory.c (hunk 2 above)
Comment (GomezGab) [reply src/commands/cmd_memory.c]: Users may expect this, but it would be incorrect. `total_node_attributes_sz_mb != sum(amortized_node_attributes_by_label_sz_mb) + amortized_unlabeled_nodes_sz_mb`. Personally, I think that the main source of confusion comes from attempting to report the memory consumption of non-disjoint labels. I think is is meaningless to say that label "A" accounts for x MB of memory when other labels may share that memory consumption.  For example:

Assume "B" is a subset of label "A". Is it accurate to say that nodes with label "A" account for 100 MB of memory and nodes of label "B"  account for 50 MB of memory, when the total taken by both "A" and  "B" is 100 MB of memory? Would it be more accurate to say that "A" accounts for 100 MB and "B" 0 MB? What about "A" 50 and "B" 50? The situation gets much more complex when you have multiple overlapping labels.

For this reason, I don't think we should break down attribute set memory usage by label. I don't think it is useful information, and it would simplify and speed up the code to not report it. However, changing this is arguably a breaking change.

File: src/graph/graph_memoryUsage.h (hunk 1 above)
Comment (GomezGab) [reply src/graph/graph_memoryUsage.h]: No, `node_attr_by_label_sz` is an array of sizes by node attribute. If every node has three attributes A, B, C, then the attr size might be 10MB for each label (which is accurate) but the total would be 10MB (not 30 MB).

Since node relations are disjoint,  `total_edge_attr_sz` is redundant, but I kept if for symmetry.
I've also considered sampling on all the edges, rather than summing the estimates per relationship type, so that `total_edge_attr_sz` would not be completely redundant (however this may be confusing so there is an argument to remove it)

**Review by swilly22 — DISMISSED** [review]

**Review by swilly22 — DISMISSED** [review]
